src/dat/controllers/CustomController.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * dat-gui JavaScript Controller Library
+ * http://code.google.com/p/dat-gui
+ *
+ * Copyright 2011 Data Arts Team, Google Creative Lab
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+import Controller from './Controller';
+
+/**
+ * @class Represents a custom controller.
+ * @param {Object} object
+ * @param {string} property
+ */
+class CustomController extends Controller{
+  constructor(object, property) {
+    super(object, property);
+
+    object.constructor( this );
```

Comment:
I'm not sure what this line is doing?

Reply:
My current version of the [CustomController](https://github.com/anhr/dat.gui/blob/CustomController/src/dat/controllers/CustomController.js) class:
```
/**
 * @class Represents a custom controller.
 * @param {init} callback function for adding of elements into this.domElement
 */
class CustomController extends Controller{
    constructor(init) {
    super({});

    init( this );
    this.custom = true;
  }
}
```
I have created my [own ES module](https://github.com/anhr/customController) for testing and use it in my [Example ](https://raw.githack.com/anhr/three.js/dev/examples/webgl_custom_controller.html). The code below is [example](https://github.com/anhr/customController/blob/master/index.js) of user's custom controller class.
```
import { GUI, controllers } from '../../dat.gui';

export class KnobController extends controllers.CustomController {
	constructor( a, b ) {
		super(function (controller) {

			var button = document.createElement('span');
			button.innerHTML = 'Knob Controller';
			button.title = 'Please press knob';
			button.style.cursor = 'pointer';
			button.style.margin = '0px 2px';
			button.onclick = function (value) {

				alert('Knob Controller ' + ( knobController.a + knobController.b ));

			}
			controller.domElement.appendChild(button);

		});
		this.a = a;
		this.b = b;
		var knobController = this;

	}
}
```